fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -776,175 +886,214 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
 	return Void();
 }
 
-// Commit one batch of transactions trs
-ACTOR Future<Void> commitBatch(
-	ProxyCommitData* self,
-	vector<CommitTransactionRequest> trs,
-	int currentBatchMemBytesCount)
-{
-	//WARNING: this code is run at a high priority (until the first delay(0)), so it needs to do as little work as possible
-	state int64_t localBatchNumber = ++self->localCommitBatchesStarted;
-	state LogPushData toCommit(self->logSystem);
-	state double t1 = now();
-	state Optional<UID> debugID;
-	state bool forceRecovery = false;
-	state int batchOperations = 0;
-	int64_t batchBytes = 0;
-	for (int t = 0; t<trs.size(); t++) {
-		batchOperations += trs[t].transaction.mutations.size();
-		batchBytes += trs[t].transaction.mutations.expectedSize();
-	}
-	state int latencyBucket = batchOperations == 0 ? 0 : std::min<int>(SERVER_KNOBS->PROXY_COMPUTE_BUCKETS-1,SERVER_KNOBS->PROXY_COMPUTE_BUCKETS*batchBytes/(batchOperations*(CLIENT_KNOBS->VALUE_SIZE_LIMIT+CLIENT_KNOBS->KEY_SIZE_LIMIT)));
+ACTOR Future<Void> preresolutionProcessingBatch(CommitBatchContext* pContext) {
 
-	ASSERT(SERVER_KNOBS->MAX_READ_TRANSACTION_LIFE_VERSIONS <= SERVER_KNOBS->MAX_VERSIONS_IN_FLIGHT);  // since we are using just the former to limit the number of versions actually in flight!
+	state CommitBatchContext& context = *pContext;
```

Comment:
Seems unnecessary, since `pContext` is just like a state variable, available for use inside this actor.